### ISSUE TITLE
Question re: scope

### ISSUE THREAD

**Issue by domchristie** [rossta/serviceworker-rails#49]
Just wondered what the reasoning was behind the scope option:

```js
navigator.serviceWorker.register('/serviceworker.js', { scope: './' })
```

With this in place, a new, separate service worker is registered for each path e.g. `/` as well as `/posts`, `/posts/new` etc.

Thanks!

**Comment by rossta** [comment]
@domchristie You are correct in that the serviceworker scope as shown will apply to those other pages; this is the default behavior when the scope option is omitted, so the choice to include it as shown is simply about making the default behavior more explicit. https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerContainer/register#Examples

 I'm not sure what you mean by new separate serviceworker being registered for each path.

**Comment by theghall** [comment]
What he means is './' causes the service worker to be registered multiple times each time a new URL is accessed.

You access domain/page1 and if you were to look at Service Workers that are loaded you would see your serviceworker associated with domain/page1.

You access domain/page2 and you would see your service worker associated with domain/page2 as well.

I am using devise and I began to see my service worker be registered with the various devise paths.

I changed the scope to '/' and it stopped happening.
